### PR TITLE
Bug 1815548 – Add offset parameter to menu2

### PR DESCRIPTION
**Pull request by mavduevskiy**
Required for [bug/1815548](https://bugzilla.mozilla.org/show_bug.cgi?id=1815548)
Adding offsets to match the UI requirements. 

unified_on_old

**LTR Vertical**
Old|New 
---|---
unified_on_old|unified_on_new
unified_off_old|unified_off_new
unified_off_old|unified_off_new
unified_off_old|unified_off_new
unified_off_old|unified_off_new

**RTL Vertical**
Old|New 
---|---
unified_on_old|unified_on_new
unified_on_old|unified_on_new
unified_on_old|unified_on_new
unified_on_old|unified_on_new

**LTR Horizontal**
Old|New 
---|---
unified_on_old|[unified_on_new
unified_on_old|unified_on_new
unified_on_old|[unified_on_new
unified_on_old|unified_on_new

**RTL Horizontal**
Old|New 
---|---
unified_on_old|unified_on_new
unified_on_old|unified_on_new
unified_on_old|unified_on_new
unified_on_old|unified_on_new

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/search/toolbar/SearchSelectorMenu.kt [hunk 1]
```diff
@@ -46,7 +48,15 @@ class SearchSelectorMenu(
         data class SearchEngine(val searchEngine: MozSearchEngine) : Item()
     }
 
-    val menuController: MenuController by lazy { BrowserMenuController() }
+    val menuController: MenuController by lazy {
+        BrowserMenuController(
+            style = MenuStyle(
+                horizontalOffset = (-4).dpToPx(context.resources.displayMetrics),
+                verticalOffset = (-2).dpToPx(context.resources.displayMetrics),
```
Comment (mavduevskiy) [inline app/src/main/java/org/mozilla/fenix/search/toolbar/SearchSelectorMenu.kt:55]: maths behind the values: SelectorMenu has the height of 28 while its container is 40 (vertical spacing is 6, target is 4) and margintStart of 8, target 4 as well. Hence, -4, -2 adjustment

Comment (t-p-white) [reply app/src/main/java/org/mozilla/fenix/search/toolbar/SearchSelectorMenu.kt:55]: Thanks for the breakdown - could we have this explanation as a comment? could save future effort in reverse engineering the maths

**Review by t-p-white — APPROVED** [review]
This looks great! Having the comment would be a nice addition, but it doesn't block this

**Comment by mavduevskiy** [comment]
thanks, @t-p-white! I updated with the comment before landing

**Pull request merged**